lib/elm-test.js

```diff
@@ -444,3 +327,202 @@ ${testFileGlobs.join('\n')}
 Are the above patterns correct? Maybe try running elm-test with no arguments?
       `.trim();
 }
+
+const parsePositiveInteger = (flag /*: string */) => (
+  string /*: string */
+) /*: number */ => {
+  const number = Number(string);
+  if (!/^\d+$/.test(string)) {
+    console.error(
+      `error: option '${flag}' expected one or more digits, but got: ${string}`
+    );
+    throw process.exit(1);
+  } else if (!Number.isFinite(number)) {
+    console.error(
+      `error: option '${flag}' expected a finite number, but got: ${number}`
+    );
+    throw process.exit(1);
+  } else {
+    return number;
+  }
+};
+
+const parseReport = (flag /*: string */) => (
+  string /*: string */
+) /*: typeof Report.Report */ => {
+  const result = Report.parse(string);
+  switch (result.tag) {
+    case 'Ok':
+      return result.value;
+    case 'Error':
+      console.error(`error: option '${flag}' ${result.error}`);
+      throw process.exit(1);
+  }
+};
+
+const examples = `
+elm-test
+  Run tests in the tests/ folder
+
+elm-test "src/**/*Tests.elm"
+  Run tests in files matching the glob
+`.trim();
+
+function main() {
+  process.title = 'elm-test';
+
+  const processes = Math.max(1, os.cpus().length);
+
+  const requireElmJsonFile = (subcommand /*: string */) => {
```

Comment:
```suggestion
  const assertElmJsonInCwd = (subcommand /*: string */) => {
```

I am probably being overzealous here, feel free to ignore me.